ci/blackbox/rdsbroker_test.go

```diff
@@ -436,17 +436,72 @@ var _ = Describe("RDS Broker Daemon", func() {
 				"postgres-micro-without-snapshot-10",
 			)
 		})
+	})
+
+	Describe("plan upgrade failure and recovery", func() {
+		TestUpdatePlan := func(serviceID, startPlanID, upgradeToPlanID, extensionName, expectedAwsTagPlanID, recoveryPlanID string) {
+			var (
+				instanceID string
+			)
+
+			BeforeEach(func() {
+				instanceID = uuid.NewV4().String()
+
+				brokerAPIClient.AcceptsIncomplete = true
+
+				code, operation, err := brokerAPIClient.ProvisionInstance(instanceID, serviceID, startPlanID, fmt.Sprintf(`{"enable_extensions": ["%s"]}`, extensionName))
+				Expect(err).ToNot(HaveOccurred())
+				Expect(code).To(Equal(202))
+				state := pollForOperationCompletion(brokerAPIClient, instanceID, serviceID, startPlanID, operation)
+				Expect(state).To(Equal("succeeded"))
+			})
+
+			AfterEach(func() {
+				brokerAPIClient.AcceptsIncomplete = true
+				code, operation, err := brokerAPIClient.DeprovisionInstance(instanceID, serviceID, startPlanID)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(code).To(Equal(202))
+				state := pollForOperationCompletion(brokerAPIClient, instanceID, serviceID, startPlanID, operation)
+				Expect(state).To(Equal("gone"))
+			})
+
+			It("handles a failure to upgrade followed by a new plan recovery", func() {
+				code, operation, err := brokerAPIClient.UpdateInstance(instanceID, serviceID, startPlanID, upgradeToPlanID, `{}`)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(code).To(Equal(202))
+				state := pollForOperationCompletion(brokerAPIClient, instanceID, serviceID, startPlanID, operation)
+				Expect(state).To(Equal("failed"))
+
+				tagPlanID, err := rdsClient.GetDBInstanceTag(instanceID, "Plan ID")
+				Expect(err).ToNot(HaveOccurred())
+				Expect(tagPlanID).To(Equal(expectedAwsTagPlanID))
+
+				code, operation, err = brokerAPIClient.UpdateInstance(instanceID, serviceID, expectedAwsTagPlanID, recoveryPlanID, `{}`)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(code).To(Equal(202))
+				state = pollForOperationCompletion(brokerAPIClient, instanceID, serviceID, startPlanID, operation)
+				Expect(state).To(Equal("succeeded"))
+
+				tagPlanID, err = rdsClient.GetDBInstanceTag(instanceID, "Plan ID")
+				Expect(err).ToNot(HaveOccurred())
+				Expect(tagPlanID).To(Equal(recoveryPlanID))
+
+			})
+		}
 
-		Describe("Postgres 10 to 11 complex failure", func() {
-			// tsearch2-caused failure will have happened after disk-space upgrade
-			// however the plan id should be rolled back even with the disk space
-			// higher than the plan
+		Describe("Postgres 10 to 11 failure followed by 10 recovery plan", func() {
+			// the test upgrades from postgres 10 to 11, which fails due to the tsearch2 extension
+			// being unavailable on postgres 11. This will leave the aws storage "off plan" with
+			// 15gb instead of 10gb.
+			// The test then moves to a recovery plan which has less disk space than the postgres 11
```

Comment:
```suggestion
			// The test then moves to another postgres 10 plan which still (in theory) has less disk space than we now actually have```

(re-flow as necessary)